static/app/components/superuserStaffAccessForm.tsx

```diff
@@ -163,21 +163,15 @@ class SuperuserStaffAccessForm extends Component<Props, State> {
     });
   };
 
-  handleLogout = async () => {
-    const {api} = this.props;
-    try {
-      await logout(api);
-    } catch {
-      // ignore errors
-    }
-    const authLoginPath = `/auth/login/?next=${encodeURIComponent(window.location.href)}`;
+  handleLogout = () => {
+    let nextUrl = `/auth/login/?next=${encodeURIComponent(window.location.href)}`;
+
     const {superuserUrl} = window.__initialData.links;
     if (window.__initialData?.customerDomain && superuserUrl) {
-      const redirectURL = `${trimEnd(superuserUrl, '/')}${authLoginPath}`;
-      window.location.assign(redirectURL);
-      return;
+      nextUrl = `${trimEnd(superuserUrl, '/')}${nextUrl}`;
```

Comment:
nit: might be safer to use the `URL` constructor rather than build a string for the url. here's a little blog on how to use the `URL` constructor and the differences between the two approaches: https://www.builder.io/blog/new-url